uaclient/security.py

```diff
@@ -497,37 +522,44 @@ def upgrade_packages_and_attach(
     For UA upgrades, prompt regarding system attach prior to upgrading UA
     packages.
     """
-    if upgrade_packages:
+    if any([upgrade_packages, upgrade_packages_ua]):
         if os.getuid() != 0:
             print(status.MESSAGE_SECURITY_APT_NON_ROOT)
             return
 
-        apt.run_apt_command(
-            cmd=["apt-get", "update"],
-            error_msg=status.MESSAGE_APT_UPDATE_FAILED,
-            print_cmd=True,
-        )
-
-        apt.run_apt_command(
-            cmd=["apt-get", "install", "--only-upgrade", "-y"]
-            + upgrade_packages,
-            error_msg=status.MESSAGE_APT_INSTALL_FAILED,
-            env={"DEBIAN_FRONTEND": "noninteractive"},
-            print_cmd=True,
-        )
-    if upgrade_packages_ua:
+    if upgrade_packages and not upgrade_packages_ua:
+        _verbose_apt_update_install(upgrade_packages)
+    else:
         if cfg.is_attached:
-            print(
-                "TODO: GH: #1402: apt commands to install missing UA updates"
-            )
+            _verbose_apt_update_install(upgrade_packages + upgrade_packages_ua)
         else:
+            cloud_type = get_cloud_type()
+            if cloud_type in PRO_CLOUDS:
+                print(
+                    status.MESSAGE_SECURITY_USE_PRO_TMPL.format(
+                        title=CLOUD_TYPE_TO_TITLE.get(cloud_type),
+                        cloud=cloud_type,
+                    )
+                )
             print(status.MESSAGE_SECURITY_UPDATE_NOT_INSTALLED_SUBSCRIPTION)
             choice = util.prompt_choices(
-                "Choose: [S]ubscribe at ubuntu.com [A]ttach"
-                " existing token [C]ancel",
+                status.PROMPT_SUBSCRIBE_OR_ATTACH,
                 valid_choices=["s", "a", "c"],
             )
-            print("TODO react to subscription choice:{} GH: #".format(choice))
+            if choice == "a":
+                print(status.PROMPT_ENTER_TOKEN)
+                token = input("> ")
+                print(
+                    "{color}{{ua attach {token}}}{end}".format(
+                        color=status.TxtColor.DISABLEGREY,
+                        token=token,
+                        end=status.TxtColor.ENDC,
+                    )
+                )
+                cli._attach_with_token(cfg, token, allow_enable=True)
```

Comment:
I think we should run the `_verbose_apt_update_install` function after attaching